### PR TITLE
cEP-0008: Review workflow at coala

### PR DESCRIPTION
**Pull request by meetmangukiya**
To track states of review comments

### CODE REVIEW

File: cEP-0008.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+know the state of a review comment. This cEP aims at forming a standard review
+system so that new reviewers can get an idea of existing review comments and
+their states.
+
+# Proposed Change
+
+To solve this problem I propose, we use the comment reactions:
+1. :+1: for **acknowledged/addressed**.
+2. :-1: for **invalid/dropped review comments**
+3. :confused: for **under discussion**.
```
Comment (sils) [inline cEP-0008.md:23]: doesn't work well for commit based reviews?

Comment (meetmangukiya) [reply cEP-0008.md:23]: One can comment individually on commits, but I see your point here. GitHub will get weird about those comments, right?

File: cEP-0008.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+| ------------ |---------------------------------------------|
+| cEP          | 8                                           |
+| Version      | 1.0                                         |
+| Title        | Review Workflow                             |
+| Authors      | Meet Mangukiya <meet123mangukiya@gmail.com> |
+| Status       | Proposed                                    |
+
+# Abstract
+
+Code Review at coala is done using GitHub pull requests. There is no feature to
```
Comment (adhikasp) [inline cEP-0008.md:13]: We have repository on gitlab too, does this applies in there?

Comment (meetmangukiya) [reply cEP-0008.md:13]: There is a resolve discussion button in gitlab,.So no need :)

File: cEP-0008.md (hunk 1 above)
Comment (Udayan12167) [reply cEP-0008.md:23]: If the code is changed the comment disappears from the apt location it will get difficult to keep track.

Comment (meetmangukiya) [reply cEP-0008.md:23]: The comment doesn't disappear, it is available as outdated diff IIRC

File: cEP-0008.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+their states.
+
+# Proposed Change
+
+To solve this problem I propose, we use the comment reactions:
+1. :+1: for **acknowledged/addressed**.
+2. :-1: for **invalid/dropped review comments**
+3. :confused: for **under discussion**.
+
+Both reviewer and author has to react to the main/top-level comment on the given
```
Comment (satwikkansal) [inline cEP-0008.md]: has to --> have to :)

File: cEP-0008.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+thread as a status report.
```
Comment (satwikkansal) [inline cEP-0008.md]: imo we can add here that "A PR may not be merged until all the review comments have been acknowledge by both the reviewer and the author"

Comment (meetmangukiya) [reply cEP-0008.md]: Yeah, we should

**Comment by meetmangukiya** [comment]
Similar thing have been proposed here as well https://github.com/coala/coala/issues/3615

File: cEP-0008.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+their states.
+
+# Proposed Change
+
+To solve this problem I propose, we use the comment reactions:
+1. :+1: for **acknowledged/addressed**.
+2. :-1: for **invalid/dropped review comments**
+3. :confused: for **under discussion**.
+
+Both reviewer and author have to react to the main/top-level comment on the
```
Comment (supergr35) [inline cEP-0008.md:25]: Potentially an example here? I'm a bit confused as to what this means myself.

Comment (meetmangukiya) [reply cEP-0008.md:25]: > Potentially an example here? I'm a bit confused as to what this means myself.

this is the _top-level comment_ in the _thread_. I ack it by :+1: ;)

Comment (meetmangukiya) [reply cEP-0008.md:25]: I am not sure how I give an example here :/

File: cEP-0008.md (hunk 2 above)
Comment (sils) [reply cEP-0008.md:13]: but then the first sentence is wrong

**Comment by sils** [comment]
this cEP should account for both github and gitlab processes.

**Comment by sils** [comment]
@meetmangukiya I'm not sure it's good to have a cEP for this, if you want this to happen why not change the docs?

**Comment by sils** [comment]
I'm not sure this'll lead to anything so closing this, if you disagree reopen.

**Pull request closed without merging**